regression/scripts/common.sh

```diff
@@ -674,6 +659,77 @@ echo "   ${subproj}: dep_jobids = ${dep_jobids}"
 echo " "
 }
 
+##----------------------------------------------------------------------------##
+# Canonicalize_filename:
+# See http://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac
+##----------------------------------------------------------------------------##
+canonicalize_filename () {
+    local target_file=$1
+    local physical_directory=""
+    local result=""
+
+    # Need to restore the working directory after work.
+    pushd `pwd` > /dev/null
+
+    cd "$(dirname "$target_file")"
+    target_file=`basename $target_file`
+
+    # Iterate down a (possible) chain of symlinks
+    while [ -L "$target_file" ]
+    do
+        target_file=$(readlink "$target_file")
+        cd "$(dirname "$target_file")"
+        target_file=$(basename "$target_file")
+    done
+
+    # Compute the canonicalized name by finding the physical path
+    # for the directory we're in and appending the target file.
+    physical_directory=`pwd -P`
+    result="$physical_directory"/"$target_file"
+
+    # restore the working directory after work.
+    popd > /dev/null
+
+    echo "$result"
+}
+
+##----------------------------------------------------------------------------##
+# Compare versions
+#
+# Example:
+#
+# EMACSVER=`emacs --version | head -n 1 | sed -e 's/.*Emacs //'`
+# if [[ `version_gt "24.0.0" $EMACSVER`  ]]; then echo yes; fi
+##----------------------------------------------------------------------------##
+function version_gt()
```

Comment:
An attempt to use `version_gt`, with the above example in a terminal, didn't seem to print `yes` for either order of the command line arguments.  Should the call be `if version_gt "24.0.0" $EMACSVER; then echo yes; fi` (this appears to work in my terminal)?

Reply:
Nice catch :exclamation:  

You are correct. I'll make that change.

Reply:
See 3cbcdfe